File: Questão-15/Exception.py
class SaldoInsuficienteException(Exception):
    def __init__(self):
        self.mensagem = "Saldo insuficiente para Sacar"
        super().__init__(self.mensagem)

class ContaBancaria:
    def __init__(self, saldo, titular):
      self.__saldo = saldo
      self.titular = titular

    def depositar(self, valor):
      self.__saldo += valor

    def sacar(self, valor):
      if self.__saldo < valor:
        raise SaldoInsuficienteException
      else:
        self.__saldo -= valor

    def get_saldo(self):
      return self.__saldo


conta = ContaBancaria(5000, "João")
conta.sacar(7000)

File: Questão-15/Exception.go
package main

import (
	"errors"
	"fmt"
)

type ContaBancaria struct{
  Titular string
  saldo float64
}

func (c *ContaBancaria)Depositar(valor float64){
  c.saldo += valor
}

func (c *ContaBancaria)Sacar(valor float64) {
  if c.saldo < valor{
    fmt.Println(errors.New("Erro: Saldo Insuficiente Para Sacar"))
	return 
  } 
  c.saldo -= valor
  }


func (c ContaBancaria)MostrarSaldo() float64{
  return c.saldo  
}




func main() {
  conta := ContaBancaria{Titular: "João", saldo: 5000}
  conta.Sacar(7000)

  
  
}